kernel/api/src/main/java/org/sakaiproject/timesheet/api/TimeSheetEntry.java

```diff
@@ -0,0 +1,49 @@
+package org.sakaiproject.timesheet.api;
+
+import java.time.Instant;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+
+import org.hibernate.annotations.Type;
+import org.sakaiproject.springframework.data.PersistableEntity;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@Entity
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+@Table(name = "TIMESHEET_ENTRY", indexes = {
+        @Index(name = "IDX_TIMESHEETENTRY_REF_TYPE", columnList = "REFERENCE")
+})
+public class TimeSheetEntry implements PersistableEntity<Long>{
+    @Id
+    @Column(name = "ID")
+    @GeneratedValue(strategy = GenerationType.AUTO, generator = "timesheet_sequence")
+    @SequenceGenerator(name = "timesheet_sequence", sequenceName = "TIMESHEET_S")
+    @EqualsAndHashCode.Include
+    private Long id;
+
+    @Column(name="REFERENCE", length = 255, nullable = false)
+    private String reference;
+
+    @Column(name = "USERID", length = 255)

```

Comment:
```suggestion
    @Column(name = "USERID", length = 99)

```
pulled length from table SAKAI_USER